aicsimageio/aics_image.py

```diff
@@ -159,20 +160,15 @@ def determine_reader(image: types.ImageLike, **kwargs: Any) -> Type[Reader]:
 
         # Try reader detection based off of file path extension
         if isinstance(image, (str, Path)):
-
-            path = str(image)
+            _, path = pathlike_to_fs(image, enforce_exists=True)
 
             # Check for extension in FORMAT_IMPLEMENTATIONS
             for format_ext, readers in FORMAT_IMPLEMENTATIONS.items():
                 if path.lower().endswith(f".{format_ext}"):
                     for reader in readers:
-                        try:
-                            ReaderClass = _load_reader(reader)
-                            if ReaderClass.is_supported_image(image):
-                                return ReaderClass
-
-                        except Exception:
-                            pass
+                        ReaderClass = _load_reader(reader)
```

Comment:
IIRC we have some file types that can be handled by multiple reader classes. This looks like it would fail at the first one, is that desired behavior? Would it make sense to log all the exceptions and then only raise if no reader has been found?

Reply:
Yep. Good call. I added a try except and log.warning the exception in the case the readers reject the path